src/frequenz/sdk/config/_manager.py

```diff
@@ -133,26 +164,72 @@ def new_receiver(
             ```
 
         Args:
+            key: The configuration key to be read by the receiver.
             skip_unchanged: Whether to skip sending the configuration if it hasn't
                 changed compared to the last one received.
 
         Returns:
             The receiver for the configuration.
         """
-        receiver = self.config_channel.new_receiver(name=str(self), limit=1)
+        receiver = self.config_channel.new_receiver(name=f"{self}:{key}", limit=1)
+
+        def _get_key_or_error(
+            config: Mapping[str, Any]
+        ) -> Mapping[str, Any] | InvalidValueForKeyError | None:
+            try:
+                return _get_key(config, key)
+            except InvalidValueForKeyError as error:
+                return error
+
+        key_receiver = receiver.map(_get_key_or_error)
 
         if skip_unchanged:
-            receiver = receiver.filter(WithPrevious(not_equal_with_logging))
+            return key_receiver.filter(WithPrevious(_not_equal_with_logging))
 
-        return receiver
+        return key_receiver
 
 
-def not_equal_with_logging(
-    old_config: Mapping[str, Any], new_config: Mapping[str, Any]
+def _not_equal_with_logging(
+    old_value: Mapping[str, Any] | InvalidValueForKeyError | None,
+    new_value: Mapping[str, Any] | InvalidValueForKeyError | None,
 ) -> bool:
     """Return whether the two mappings are not equal, logging if they are the same."""
-    if old_config == new_config:
+    if old_value == new_value:
         _logger.info("Configuration has not changed, skipping update")
-        _logger.debug("Old configuration being kept: %r", old_config)
         return False
+
+    if isinstance(new_value, InvalidValueForKeyError) and not isinstance(
+        old_value, InvalidValueForKeyError
+    ):
+        _logger.error(
+            "Configuration for key %r has an invalid value: %r",
+            new_value.key,
+            new_value.value,
+        )
     return True
+
+
+def _get_key(config: Mapping[str, Any], key: str) -> Mapping[str, Any] | None:
+    """Get the value from the configuration under the specified key.
+
+    Args:
+        config: The configuration to get the value from.
+        key: The key to get the value for.
+
+    Returns:
+        The value under the key, or `None` if the key is not found.
+
+    Raises:
+        InvalidValueForKeyError: If the value under the key is not a mapping.
+    """
+    match config.get(key):
```

Comment:
It can raise KeyError, is it ok?
If yes, could you please add comment why as it is not obvious to me :) 
If no, Maybe raising InvalidValueForKeyError for `KeyError`?